source/common/http/utility.cc

```diff
@@ -997,6 +997,7 @@ RequestMessagePtr Utility::prepareHeaders(const envoy::config::core::v3::HttpUri
   RequestMessagePtr message(new RequestMessageImpl());
   message->headers().setPath(path);
   message->headers().setHost(host);
+  message->headers().setUserAgent(Headers::get().UserAgentValues.GoBrowser);
```

Comment:
this gets called from a variety of places, I suggest we move this closer to where this header is actually needed e.g. in jwk_fetcher itself?

Reply:
Sure I can do that. I presumed it won't break but since you have better picture I will do it 


Tho I can't set the testing environment in my side

Reply:
Here are general instructions for building and running tests: https://github.com/envoyproxy/envoy/blob/main/bazel/README.md

You can also use the CI below to "test" your changes since it has the build env set up -- though it'll take a lot longer than just running a unit test locally.

Reply:
I followed the instructions, a shared object file is missing when i try to setup clang for bazel. I will try to later but CI should be suffice i guess